pkg/controllers/capr/machineprovision/controller.go

```diff
@@ -384,7 +384,6 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 			// passed here, it will delete all objects it finds.
 			return obj, h.apply.WithOwner(obj).ApplyObjects()
 		}
-		return obj, generic.ErrSkip
```

Comment:
We skip this because the `h.doRemove` logic still needs to be handled if it is false for the job to be marked false (for some reason, doesn't make sense to me but it breaks if its skips).

Reply:
If I leave it, the job will endlessly spew pods, which betrays my expectations and I don't believe is the preferred behavior.

Reply:
Also, it never affects the case where condition is `True`, since we would never return a `generic.Errskip`, we always return `h.apply.WithOwner(obj).ApplyObjects()` to remove the job.

Reply:
There is an edge case that is now not caught with the removal of the ErrSkip return. If the delete job pod has succeeded but the job controller has not yet reconciled completion time on the job object from the pod status, `shouldCleanupObjects` will be false and we'll continue on execution. I think it might make sense to add a conditional check if `cond == "True"` then return the `ErrSkip` so we don't break existing behavior in that sense.

Reply:
The deleteJobCondition is set based on job pod status, whereas `shouldCleanupObjects` works off of the job status for completion time.